Runtime/Components/Cinemachine3rdPersonFollow.cs

```diff
@@ -139,56 +158,103 @@ void PositionCamera(ref CameraState curState, float deltaTime)
+            var shoulderPivotReflected = Vector3.Reflect(ShoulderOffset, Vector3.right);
+            var shoulderOffset = Vector3.Lerp(shoulderPivotReflected, ShoulderOffset, CameraSide);
+            t_HandOffset.y = VerticalArmLength;
+            shoulder = root + Quaternion.AngleAxis(m_PreviousHeadingAngle, Vector3.up) * shoulderOffset;
+            hand = shoulder + FollowTargetRotation * t_HandOffset;
         }
+        Vector3 t_HandOffset = Vector3.zero; // minor opt. - to avoid creating a new vector in GetRigPositions
 
-        private Vector3 PullTowardsStartOnCollision(
+        bool PullTowardsStartOnCollision(
             in Vector3 rayStart, in Vector3 rayEnd,
-            in LayerMask filter, float radius)
+            in LayerMask filter, float radius,
+            out Vector3 result)
         {
             var dir = rayEnd - rayStart;
             bool hasHit = RuntimeUtility.SphereCastIgnoreTag(
                 rayStart, radius, dir, out RaycastHit hitInfo, dir.magnitude, filter, IgnoreTag);
-            return hasHit ? hitInfo.point + hitInfo.normal * radius: rayEnd;
+            result = hasHit ? hitInfo.point + hitInfo.normal * radius: rayEnd;
+            return hasHit;
+        }
+        
+        
+        /// <summary>
+        /// Pulls camera back to its normal position when not colliding. The pull back speed is determined by
+        /// the CollisionDamping class member and deltaTime.
+        /// </summary>
+        /// <param name="isColliding">Is the camera currently colliding?</param>
+        /// <param name="deltaTime">Used for damping.</param>
+        /// <param name="root">Root of the rig</param>
+        /// <param name="resolved">Position of the camera.</param>
+        /// <param name="previousDistance">Distance of the camera from root in the previous frame.</param>
+        /// <returns>Damped position</returns>
+        Vector3 DampedPullBackPostCollision(bool hasHit, float deltaTime, 
+            Vector3 root, Vector3 resolved, ref bool prevHasHit, ref bool postCollision, ref float previousDistance)
+        {
+            if (!hasHit && prevHasHit)
+                postCollision = true;
+            else if (hasHit)
+                postCollision = false;
+            
+            prevHasHit = hasHit;
+            // Post correction damping without rotational damp
+            var dampedResolved = resolved;
+            if (postCollision && CollisionDamping > 0 && deltaTime >= 0)
+            {
+                Vector3 difference = resolved - root;
+                float delta = difference.magnitude - previousDistance;
+                delta = Damper.Damp(delta, CollisionDamping, deltaTime);
+                dampedResolved = root + difference.normalized * (delta + previousDistance);
+
+                postCollision = (resolved - dampedResolved).sqrMagnitude >= UnityVectorExtensions.Epsilon;
+            }
+            previousDistance = (dampedResolved - root).magnitude;
```

Comment:
this code is very inefficient - lots of sqrts being calculated, when you only need to do it once